ioztat

```diff
@@ -76,12 +76,68 @@ def parseDatasetsLinux(pool):
             datasets[dataset.name] = dataset
     return datasets
 
+### FREEBSD Support
+#
+#
+class DatasetFreeBSD(DatasetBase):
+    def __init__(self, objset_id = None):
+        """
+        Prepare for reading sysctl data for the dataset with the given id.
+        The data is provided by sysctl, passed later to consume()
+        """
+        self.objset_id = objset_id
+
+    def consume(self, field, value):
+        if field == 'dataset_name':
+            self.name = value
+        if field == 'writes':
+            self.writes = int(value)
+        if field == 'nwritten':
+            self.nwritten = int(value)
+        if field == 'reads':
+            self.reads = int(value)
+        if field == 'nread':
+            self.nread = int(value)
+
+def parseDatasetsFreeBSD(pool):
+    zfsDatasetKStatPrefix = "kstat.zfs.{!s}.dataset".format(pool)
+    # Collect by objset_id, first
+    objsets = {}
+    r = subprocess.run(["sysctl", zfsDatasetKStatPrefix], capture_output=True, check=True, text=True)
+    now = time.monotonic_ns()
+    current_objset = None
+    for line in r.stdout.split("\n"):
+        if not line.strip():
+            continue
+        # TODO: can a dataset name contain a ':' ?
```

Comment:
Yes.  Alphanumerics plus `-`, `_`, `:` and `.`.   You could make this a `split(":", 2)`, or switch to `=`  with the `-e` flag.

Reply:
`-e` seems like a good idea, thank you for the suggestion.